src/controllers/localWebService.ts

```diff
@@ -33,10 +36,26 @@ export default class LocalWebService {
         this.app.set('view engine', 'ejs');
         Utils.logDebug(Constants.msgLocalWebserviceStaticContent + LocalWebService.staticContentPath);
         this.server.on('request', this.app);
+
+        // Handle new connections to the web socket server
         this.wss.on('connection', (ws) => {
             let parse = querystring.parse(url.parse(ws.upgradeReq.url).query);
-            self.clientMap.set(parse.uri, ws);
-            self.newConnection.emit('connection', parse.uri);
+            if (self.wsMap.has(parse.uri)) {
```

Comment:
Have you tested this with multiple result windows and switching between them? I suspect this won't work.